tools/src/main/java/org/apache/kafka/tools/ToolsUtils.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.tools;
+
+import joptsimple.OptionParser;
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.server.util.CommandLineUtils;
+
+import java.util.Arrays;
+
+public class ToolsUtils {
+    public static void validatePortOrExit(OptionParser parser, String hostPort) {
```

Comment:
I know you just translated this to Java, but I think we can add null/empty checks for the input parameters.

Reply:
And what should it do when null?
Could you also tell me more about improvement? I can't find FQCN in my implementation.

Reply:
> And what should it do when null?

The parser can be null right? We don't want to throw a NPE to the user. You can print an error in that case and exit.

The hostPort can be null or empty. Same story.

> Could you also tell me more about improvement? I can't find FQCN in my implementation. And how it must be corrected?

It's in tests/kafkatest/services/kafka/kafka.py, where you already changed the package name. My suggestion is to use the wrapper script, instead of the FQCN. That way, if the package name needs to be changed again in the future, we won't need to touch the system test.

Reply:
Do you have ideas about how it should look like? I see there 3 FQCN. What's the way to put this names in that script?

Reply:
There are only a couple of FQCN used in tests/kafkatest/services/kafka/kafka.py, but we are only concerned about kafka-get-offsets, the other will be addressed in its own PR. The change is very simple in the end:

```sh
diff --git a/tests/kafkatest/services/kafka/kafka.py b/tests/kafkatest/services/kafka/kafka.py
index f116f92c3b..ab48d1091d 100644
--- a/tests/kafkatest/services/kafka/kafka.py
+++ b/tests/kafkatest/services/kafka/kafka.py
@@ -1777,8 +1777,7 @@ class KafkaService(KafkaPathResolverMixin, JmxMixin, Service):
         node = self.nodes[0]
 
         cmd = fix_opts_for_new_jvm(node)
-        cmd += self.path.script("kafka-run-class.sh", node)
-        cmd += " org.apache.kafka.tools.GetOffsetShell"
+        cmd += self.path.script("kafka-get-offsets.sh", node)
         cmd += " --bootstrap-server %s" % self.bootstrap_servers(self.security_protocol)
 
         if time:
```

Reply:
Ah, so you mean it should be just moved from kafka/tests/core to kafka/tests/tools?

Reply:
No, that's a service class that includes multiple tools.

Reply:
@ruslankrivoshein let me know when you have time to apply that patch, so that I can do the final review. Thanks.

Reply:
@fvaleri when I add there null checking I face this situation. Should I really add it or this fail must be worked around?
```
Verification failed: SpotBugs ended with exit code 1
NP_NULL_ON_SOME_PATH: Possible null pointer dereference
```

Reply:
I don't see the code, so I can't comment further, but this reveals a programming error.

Reply:
@fvaleri ah, I'm sorry, I investigated it finally and worked around.
What do you think about this approach?
```
public static void validatePortOrExit(OptionParser parser, String hostPort) {
    if (parser == null || hostPort == null || hostPort.isEmpty()) {
        CommandLineUtils.printVersionAndExit();
        return;
    }
}
```
Should I use printUsageAndExit if parser will be null or what will be better to indicate an error?

Reply:
@fvaleri what do you think, do we really need here "parser"? What about removing usage of CommandLineUtils and just to print message into stderr?

Reply:
Yes, I think that's a reasonable approach.